main.py

```diff
@@ -105,7 +117,16 @@
                 time.sleep(1)
 
                 # reset state
+                # TODO check if subgrid needs to be reset
+                previous_subgrid = state.user_select_subgrid
                 state = UserTermState()
+
+                # check if subgrid needs to be updated
+                if state.save_subgrid:
+                    state.user_select_subgrid = previous_subgrid
```

Comment:
Shouldn't this be saving `state.user_select_space` to `state.user_select_subgrid` (assuming that subgrid is valid which we will implement later?